internal/provider/waypoint/data_source_waypoint_application.go

```diff
@@ -100,8 +120,28 @@ func (d *DataSourceApplication) Configure(ctx context.Context, req datasource.Co
 	d.client = client
 }
 
+// ApplicationDataSourceModel describes the data source data model
+type ApplicationDataSourceModel struct {
+	ID                      types.String `tfsdk:"id"`
+	Name                    types.String `tfsdk:"name"`
+	ProjectID               types.String `tfsdk:"project_id"`
+	OrgID                   types.String `tfsdk:"organization_id"`
+	ReadmeMarkdown          types.String `tfsdk:"readme_markdown"`
+	ApplicationTemplateID   types.String `tfsdk:"application_template_id"`
+	ApplicationTemplateName types.String `tfsdk:"application_template_name"`
+	NamespaceID             types.String `tfsdk:"namespace_id"`
+
+	// deferred for now
+	// Tags       types.List `tfsdk:"tags"`
+
+	// deferred and probably a list or objects, but may possible be a separate
+	// ActionCfgs types.List `tfsdk:"action_cfgs"`
+
+	InputVars types.Set `tfsdk:"app_input_vars"`
+}
+
 func (d *DataSourceApplication) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
-	var data ApplicationResourceModel
+	var data ApplicationDataSourceModel
```

Comment:
Why the change here? The underlying data model remains the same; the type that backs the `types.Set` here is the same as in the `resource_application`

Reply:
I see the data source has just input vars and the resource now has TemplateInputVars as well. Why the discrepancy?

Reply:
The discrepancy is because of the distinction that we recently discussed regarding splitting variables set on a template by the platform engineer into one "bucket", and variables chosen by the platform engineer to another. The data source cannot make the distinction because the input variables are not in the TF config of the data source (since it's intended to read that info from HCP Waypoint).